assets/app/data/crowbar/services/upgrade-status.factory.spec.js

```diff
@@ -187,6 +187,61 @@ describe('Upgrade Status Factory', function () {
                     });
                 });
             });
+
+            describe('when got upgrade status from api unsuccessfully', function () {
+                describe('when timeout == 0', function () {
+                    beforeEach(function () {
+                        bard.mockService(upgradeFactory, {
+                            getStatus: $q.reject()
+                        });
+
+                        upgradeStatusFactory.waitForStepToEnd(
+                            testedStep, mockedSuccessCallback, mockedErrorCallback, pollingInterval
+                        );
+
+                        $rootScope.$digest();
+                    });
+
+                    it('should not call success callback', function () {
+                        expect(mockedSuccessCallback).not.toHaveBeenCalled();
+                    });
+
+                    it('should call error callback', function () {
+                        expect(mockedErrorCallback).toHaveBeenCalled();
+                    });
+
+                    it('should not schedule another check', function () {
+                        expect(mockedTimeout).not.toHaveBeenCalled();
+                    });
+                });
+
+                describe('when timeout > 0', function () {
+                    beforeEach(function () {
+                        bard.mockService(upgradeFactory, {
+                            getStatus: $q.reject()
+                        });
+
+                        upgradeStatusFactory.waitForStepToEnd(
+                            testedStep, mockedSuccessCallback, mockedErrorCallback, 1, 10
+                        );
+
+                        $rootScope.$digest();
+                    });
+
+                    it('should not call success callback', function () {
+                        expect(mockedSuccessCallback).not.toHaveBeenCalled();
+                    });
+
+                    it('should not call error callback', function () {
+                        expect(mockedErrorCallback).not.toHaveBeenCalled();
+                    });
+
+                    it('should schedule another check', function () {
+                        expect(mockedTimeout).toHaveBeenCalled();
```

Comment:
Is it possible to assert the number of times the function is being called?
Also, the two parameters that are being sent.